core/Config/IniFileChain.php

```diff
@@ -538,4 +543,22 @@ private function dumpSettings($values, $header)
         $writer = new IniWriter();
         return $writer->writeToString($values, $header);
     }
+
+    private function replaceInvalidChars($value)
+    {
+        if (is_array($value)) {
+            $result = [];
+            foreach ($value as $key => $arrayValue) {
+                $key = $this->replaceInvalidChars($key);
+                if (is_array($arrayValue)) {
+                    $arrayValue = $this->replaceInvalidChars($arrayValue);
+                }
+
+                $result[$key] = $arrayValue;
+            }
+            return $result;
+        } else {
+            return preg_replace('/[^a-zA-Z0-9_-]/', '', $value);
```

Comment:
That wouldn't solve this problem, because `[...]` is valid in this case, but `]asldkjf`, eg, is not. Throwing an exception isn't an issue for me though so if that is more desired than removing the chars then that's fine too.